pkg/istio/authorizationpolicy/controller.go

```diff
@@ -111,11 +113,16 @@ func (c *Controller) processEvent(fn cache.KeyFunc, obj interface{}) {
 
 // Run starts the main controller loop running sync at every poll interval.
 func (c *Controller) Run(stopCh <-chan struct{}) {
-	go c.resync(stopCh)
+	if c.standAloneMode {
+		go c.serviceIndexInformer.Run(stopCh)
+		go c.configStoreCache.Run(stopCh)
+		go c.adIndexInformer.Run(stopCh)
+	}
 
 	if !cache.WaitForCacheSync(stopCh, c.configStoreCache.HasSynced, c.serviceIndexInformer.HasSynced, c.adIndexInformer.HasSynced) {
 		log.Panicln("Timed out waiting for namespace cache to sync.")
 	}
+	go c.resync(stopCh)
```

Comment:
How does the behavior change if we resync after cache-sync instead of before?

Reply:
In the previous version, the authorizationpolicy controller was instantiated by the controller (https://github.com/yahoo/k8s-athenz-istio-auth/blob/master/pkg/controller/controller.go#L154-L157)[here], the values like  `c.configStoreCache` , `c.serviceIndexInformer` and `c.adIndexInformer` are shared between these controllers. Before the `Run` from the autorizatioinpolicy controller was called, it waits for cache sync (https://github.com/yahoo/k8s-athenz-istio-auth/blob/master/pkg/controller/controller.go#L218-L227)[here]. 

As the we already waited on the cache sync in the controller already, the cache sync here is redundant for the authorization controller.